cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactInspector.java

```diff
@@ -188,6 +198,91 @@ public ArtifactClasses inspectArtifact(Id.Artifact artifactId, File artifactFile
   }
 
   /**
+   * Inspects the plugin and once we find the plugin with the interested pluginMethod to invoke,
+   * we instantiate the plugin and invoke the plugin,
+   * we deserialize the request body into parameter type and pass it when invoking the plugin method and also
+   * optionally pass {@link EndpointPluginContext} to the method if its declared in the plugin method.
+   * @param artifactFile artifact file of the plugin
+   * @param artifact the id of the plugin artifact
+   * @param pluginInstantiator to instantiate plugin instances.
+   * @param pluginType type of the plugin
+   * @param pluginName name of the plugin
+   * @param pluginMethod plugin method to invoke
+   * @param methodParameter parameter body
+   * @param endpointPluginContext endpoint plugin context that can optionally be passed to the method
+   * @throws IOException if there was an exception opening the jar file
+   * @throws InvalidArtifactException if the artifact is invalid
+   * @throws NotFoundException Not Found exception thrown from plugin method invocation or if matching plugin is found.
+   */
+  public String inspectPluginsAndInstantiatePluginClass(File artifactFile, Id.Artifact artifact,
+                                                        PluginInstantiator pluginInstantiator,
+                                                        String pluginType, String pluginName, String pluginMethod,
+                                                        String methodParameter,
+                                                        DefaultEndpointPluginContext endpointPluginContext)
+    throws IOException, InvalidArtifactException, NotFoundException {
+
+    // See if there are export packages. Plugins should be in those packages
+    Set<String> exportPackages = getExportPackages(artifactFile);
+
+    ClassLoader pluginClassLoader = pluginInstantiator.getArtifactClassLoader(artifact.toArtifactId());
```

Comment:
Seems like there are some code duplication with existing code base. We already have code to inspect plugin jar and have code to instantiate plugin instance. Can you refactor?